flux-fixpoint/src/constraint.rs

```diff
@@ -517,6 +543,95 @@ impl fmt::Display for Constant {
 impl Constant {
     pub const ZERO: Constant = Constant::Int(Sign::Positive, 0);
     pub const ONE: Constant = Constant::Int(Sign::Positive, 1);
+
+    fn to_bool(&self) -> Option<bool> {
+        match self {
+            Constant::Bool(b) => Some(*b),
+            _ => None,
+        }
+    }
+
+    fn to_int(&self) -> Option<i128> {
+        match self {
+            Constant::Int(Sign::Positive, n) => Some(*n as i128),
```

Comment:
This is wrong because `n as i128` can overflow. The reason ints are represented with a sign and a `u128` and not just an `i128`, is that we need to be able to represent both `u128::MAX` and `i128::MIN` as they are both valid rust literals. 

Initially, I thought about representing numbers with [BigInt](https://docs.rs/big_num/latest/big_num/struct.BigInt.html) since at this level we are actually dealing with mathematical integers, but I ended up going for the simpler representation because we only needed to represent valid rust literals. If we are going to do arithmetic maybe we need to reconsider (or be very careful with overflows).